contracts/strategies/StrategyCreamCRV.sol

```diff
@@ -0,0 +1,193 @@
+        if (_balance < _amount) {
+            _amount = _withdrawSome(_amount.sub(_balance));
+            _amount = _amount.add(_balance);
+        }
+
+        uint256 _fee = _amount.mul(withdrawalFee).div(withdrawalMax);
+
+        IERC20(want).safeTransfer(IController(controller).rewards(), _fee);
+        address _vault = IController(controller).vaults(address(want));
+        require(_vault != address(0), "!vault"); // additional protection so we don't burn the funds
+
+        IERC20(want).safeTransfer(_vault, _amount.sub(_fee));
+    }
+
+    // Withdraw all funds, normally used when migrating strategies
+    function withdrawAll() external returns (uint256 balance) {
+        require(msg.sender == controller, "!controller");
+        _withdrawAll();
+
+        balance = IERC20(want).balanceOf(address(this));
+
+        address _vault = IController(controller).vaults(address(want));
+        require(_vault != address(0), "!vault"); // additional protection so we don't burn the funds
+        IERC20(want).safeTransfer(_vault, balance);
+    }
+
+    function _withdrawAll() internal {
+        uint256 amount = balanceC();
+        if (amount > 0) {
+            _withdrawSome(balanceCInToken().sub(1));
+        }
+    }
+
+    function harvest() public {
+        require(msg.sender == strategist || msg.sender == governance, "!authorized");
+        Creamtroller(creamtroller).claimComp(address(this));
+        // NOTE: in case of CREAM liquidity mining
+        uint256 _cream = IERC20(cream).balanceOf(address(this));
+        if (_cream > 0) {
+            IERC20(cream).safeApprove(uni, 0);
+            IERC20(cream).safeApprove(uni, _cream);
+
+            address[] memory path = new address[](3);
+            path[0] = cream;
+            path[1] = weth;
+            path[2] = want;
+
+            Uni(uni).swapExactTokensForTokens(_cream, uint256(0), path, address(this), now.add(1800));
+        }
+        uint256 _want = IERC20(want).balanceOf(address(this));
+        if (_want > 0) {
+            uint256 _fee = _want.mul(performanceFee).div(performanceMax);
+            IERC20(want).safeTransfer(IController(controller).rewards(), _fee);
+            deposit();
+        }
+    }
+
+    function _withdrawSome(uint256 _amount) internal returns (uint256) {
+        uint256 b = balanceC();
+        uint256 bT = balanceCInToken();
+        // can have unintentional rounding errors
+        uint256 amount = (b.mul(_amount)).div(bT).add(1);
+        uint256 _before = IERC20(want).balanceOf(address(this));
+        _withdrawC(amount);
+        uint256 _after = IERC20(want).balanceOf(address(this));
+        uint256 _withdrew = _after.sub(_before);
+        return _withdrew;
+    }
+
+    function balanceOfWant() public view returns (uint256) {
+        return IERC20(want).balanceOf(address(this));
+    }
+
+    function _withdrawC(uint256 amount) internal {
+        cToken(crCRV).redeem(amount);
```

Comment:
Redeem doesn't revert with an error code if insufficient funds, require(success...

Reply:
added the require check on latest on commit 018df91, tested against specific amounts but still need more testing on `vault.withdrawAll` to confirm it works for all cases.

Reply:
latest tests adds testing for withdraws from vault and from strategy